src/realm/util/serializer.cpp

```diff
@@ -164,16 +165,24 @@ std::string print_value<>(realm::ObjKey k)
     return ss.str();
 }
 
-template <>
-std::string print_value<>(realm::ObjLink link)
+std::string print_value(realm::ObjLink link, Group* g)
 {
-    std::stringstream ss;
     if (!link) {
-        ss << "NULL";
+        return "NULL";
     }
     else {
-        ss << "L" << link.get_table_key().value << ":" << link.get_obj_key().value;
+        TableRef target_table = g->get_table(link.get_table_key());
```

Comment:
Let's add a null check on the group and fallback to the "Lx:y" syntax if not found.